java/java.mx.project/src/org/netbeans/modules/java/mx/project/SuiteSources.java

```diff
@@ -0,0 +1,1195 @@
+            arr.add(library);
+            fillDeps.put(library.getName(), library);
+        }
+        return arr;
+    }
+
+    private static Map<String, SuiteSources> findImportedSuites(FileObject dir, MxSuite s, Map<String, Dep> fillDeps) {
+        if (dir == null) {
+            return Collections.emptyMap();
+        }
+        CORE.registerDeps("mx", fillDeps);
+        final MxImports imports = s.imports();
+        if (imports != null) {
+            Map<String, SuiteSources> imported = new LinkedHashMap<>();
+            for (MxImports.Suite imp : imports.suites()) {
+                SuiteSources impSources = findSuiteSources(dir, imp);
+                final String suiteName = imp.name();
+                if (impSources == null) {
+                    LOG.log(Level.INFO, "cannot find imported suite: {0}", suiteName);
+                    continue;
+                }
+                imported.put(suiteName, impSources);
+                impSources.registerDeps(suiteName, fillDeps);
+            }
+            return imported;
+        }
+        return Collections.emptyMap();
+    }
+
+    private List<Dist> findDistributions(MxSuite s, List<Library> libraries, List<Group> groups, Map<String, Dep> fillDeps) {
+        List<Dist> dists = new ArrayList<>();
+        for (Map.Entry<String, MxDistribution> entry : s.distributions().entrySet()) {
+            Dist d = new Dist(entry.getKey(), entry.getValue());
+            dists.add(d);
+            fillDeps.put(d.getName(), d);
+        }
+        return dists;
+    }
+
+    final synchronized void computeTransitiveDeps() {
+        Map<String, Dep> collectedDeps = this.transitiveDeps;
+        if (collectedDeps == null) {
+            return;
+        }
+        this.transitiveDeps = null;
+        for (Library l : this.libraries) {
+            transitiveDeps(l, collectedDeps);
+        }
+        for (Group g : this.groups) {
+            transitiveDeps(g, collectedDeps);
+        }
+        for (Dist d : this.distributions) {
+            transitiveDeps(d, collectedDeps);
+        }
+        for (Group g : groups) {
+            g.computeClassPath(collectedDeps);
+        }
+        for (Dist d : this.distributions) {
+            d.computeSourceRoots(collectedDeps);
+        }
+    }
+
+    private static SuiteSources findSuiteSources(FileObject dir, MxImports.Suite imp) throws IllegalArgumentException {
+        SuiteSources sources = findSuiteSources(dir.getParent(), imp.name());
+        if (sources != null) {
+            return sources;
+        }
+        if (imp.subdir()) {
+            for (FileObject subDir : dir.getParent().getChildren()) {
+                sources = findSuiteSources(subDir, imp.name());
+                if (sources != null) {
+                    return sources;
+                }
+            }
+            for (FileObject subDir : dir.getParent().getParent().getChildren()) {
```

Comment:
Is the 2-level parent limit hardocded in `mx` as well ?

Reply:
The existing sources that I am aware of are find with 2-level limit. I am not sure if the 2-level parent limit is in the {{mx}} itself or not. Scanning more deeply might have negative effect on performance. Let's leave it as it is unless 
@gilles-duboscq wants to comment.

Reply:
There is indeed a 2-level limit in mx: starting from the directory in which the various repositories are checked out, the "importer" specifies if the imported suite is found at depth 1 or 2, no other depth can be specified (default is depth 1 and a `"subdir" : True` attribute switches the depth to 2).